controller/monitor/snapshot_monitor.go

```diff
@@ -332,6 +332,17 @@ func (m *SnapshotMonitor) run(arg interface{}) error {
 		return errors.Wrapf(err, "failed to get engine for volume %v", task.volumeName)
 	}
 
+	// Skip snapshot hashing if the volume has less than 2 replicas
+	volume, err := m.ds.GetVolumeRO(engine.Spec.VolumeName)
+	if err != nil {
+		m.logger.WithField("monitor", monitorName).WithError(err).Debugf("failed to get volume %v for snapshot calculation", engine.Spec.VolumeName)
+		return nil
+	}
+	if volume.Spec.NumberOfReplicas <= 1 {
```

Comment:
Can we check the healthy replica count instead?

Reply:
Addressed. Thank you.